### ISSUE TITLE
replace einsum shuffling in tensorviewer with variable indexiing

### ISSUE THREAD

**Issue by lukasheinrich** [diana-hep/pyhf#594]
# Description

I just realized that the tensor backends support indexing like 

```
a = torch.randn(1,2,3)
a[...,0]
a[...,1]
a[...,2]
```

to access the last axis.. in light of this probably the einsum shuffling in `TensorViewer` can be replaced

**Comment by kratsg** [comment]
Why or how is this different from `a[:,:,0]` and `a[:,:,1]` and so on? I guess the `...` means an arbitrary number of dimensions? Is there a performance difference? Maybe `a[...,N] = a.T[N]`

**Comment by lukasheinrich** [comment]
Yeah it’s the same as the colon version just for arbitrary number of
parameters, I could see that this does not change the memory layout and
this should be faster

**Comment by kratsg** [comment]
> I could see that this does not change the memory layout and this should be faster

This is new to me (surprisingly!) but faster than the transpose version? Or than the colon access?
